Update README and correct lint error (#1)
auth0/express-oauth2-bearer: README.md

> Authentication middleware for Express.js that validates access tokens following [RFC 6750](https://tools.ietf.org/html/rfc6750). The purpose of this library is to protect Oauth2 resources.

Reviewer: Small suggestion. The RFC refers to it as "OAuth 2.0". 

```suggestion
Authentication middleware for Express.js that validates access tokens following [RFC 6750](https://tools.ietf.org/html/rfc6750). The purpose of this library is to protect OAuth 2.0 resources.
```

**Please Note:** This library is currently in pre-release status and has not had a complete security review. We **do not** recommend using this library in production yet. As we move towards early access, please be aware that releases may contain breaking changes. We will be monitoring the Issues queue here for feedback and questions. PRs and comments on existing PRs are welcome!

## Table of Contents

- [Installation](#installation)
- [Getting Started](#getting-started)
- [Contributing](#contributing)
- [Support + Feedback](#support--feedback)
- [Vulnerability Reporting](#vulnerability-reporting)
- [What is Auth0](#what-is-auth0)
- [License](#license)

## Installation

This library is installed with [npm](https://npmjs.org/package/express-oauth2-bearer):

```
npm i express-oauth2-bearer --save
```

## Getting Started

The library needs the following values to authroize requests:

- **Issuer Base URL**: The base URL of the authorization server. If you're using Auth0, this is your tenant **Domain** pre-pended with `https://` (like `https://tenant.auth0.com`) found on the **Settings** tab for your Application in the [Auth0 dashboard](https://manage.auth0.com).

> - **Allowed Audiences**: Audience identifier (or mulitple separated by a comma) allowed for the access token. If you're using Auth0, this is the **Identifier** found on the **Settings** tab for your API in the [Auth0 dashboard](https://manage.auth0.com/#/apis).

Reviewer: ```suggestion
- **Allowed Audiences**: Audience identifier (or multiple separated by a comma) allowed for the access token. If you're using Auth0, this is the **Identifier** found on the **Settings** tab for your API in the [Auth0 dashboard](https://manage.auth0.com/#/apis).
```

  });
```

### API Documentation:

> `auth()` accepts an asynchronous function receiving a token and returning a set of claims.

Reviewer: is it worth clarifying _which_ token?

Author: ```suggestion
`auth()` accepts an asynchronous function receiving an access token and returning a set of claims.
```

> `requiredScopes()` accepts either an string or an array of strings.

Reviewer: ```suggestion
`requiredScopes()` accepts either a string or an array of strings.
```

`strategies.openid` accepts the following parameters:

| Name                | Default                            | Description                                                          |
|---------------------|------------------------------------|----------------------------------------------------------------------|
| issuerBaseURL       | `env.ISSUER_BASE_URL`              | URL for the token issuer.                                            |
| allowedAudiences    | `env.ALLOWED_AUDIENCES.split(',')` | Allowed audiences for the token.                                     |

> | clockTolerance      | `5`                                | Clock tolerance in seconds for token verification, aka leeway.       |

Reviewer: Any particular reason to use 5 secs over other value? e.g. auth0-spa-js default leeway is 60.

Author: That's just what it came with. Would need to update it in the code as well. We'll make that part of the EA criteria.

> | clientSecret        | `env.CLIENT_SECRET`                | Cleitn secret, required for tokens signed with symmetric algorithms. |

Reviewer: ```suggestion
| clientSecret        | `env.CLIENT_SECRET`                | Client secret, required for tokens signed with symmetric algorithms. |
```

## Contributing

We appreciate feedback and contribution to this repo! Before you get started, please see the following:

- [Auth0's general contribution guidelines](https://github.com/auth0/.github/blob/master/CONTRIBUTING.md)
- [Auth0's code of conduct guidelines](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md)

Contributions can be made to this library through PRs to fix issues, improve documentation or add features. Please fork this repo, create a well-named branch, and submit a PR with a complete template filled out.

> Code changes in PRs should be accompanied by tests covering the changed or added functionality. Tests can be run for this library with:

Reviewer: Maybe this should be part of a `DEVELOPING` file in the root

Author: At some point, maybe ... just wanted to get this information in there for now.

```bash
npm install
npm test
```

When you're ready to push your changes, please run the lint command first:

```bash
npm run lint
```

## Support + Feedback

> Please use the [Issues queue](https://github.com/auth0/express-openid-connect/issues) in this repo for questions and feedback.

Reviewer: wrong repo link?

```suggestion
Please use the [Issues queue](https://github.com/auth0/express-oauth2-bearer/issues) in this repo for questions and feedback.
```
